python/ray/tune/suggest/optuna.py

```diff
@@ -147,3 +186,66 @@ def restore(self, checkpoint_path):
             save_object = pickle.load(inputFile)
         self._storage, self._pruner, self._sampler, \
             self._ot_trials, self._ot_study = save_object
+
+    @staticmethod
+    def convert_search_space(spec: Dict):
+        spec = flatten_dict(spec, prevent_delimiter=True)
+        resolved_vars, domain_vars, grid_vars = parse_spec_vars(spec)
+
+        if not domain_vars and not grid_vars:
+            return []
+
+        if grid_vars:
+            raise ValueError(
+                "Grid search parameters cannot be automatically converted "
+                "to an Optuna search space.")
+
+        def resolve_value(par, domain):
+            quantize = None
+
+            sampler = domain.get_sampler()
+            if isinstance(sampler, Quantized):
+                quantize = sampler.q
+                sampler = sampler.sampler
+
+            if isinstance(domain, Float):
+                if isinstance(sampler, LogUniform):
+                    if quantize:
+                        logger.warning(
+                            "Optuna does not support both quantization and "
+                            "sampling from LogUniform. Dropped quantization.")
```

Comment:
hm is there a reason why we don't raise here instead?

Reply:
IMO quantization is more a cosmetic thing. The resulting parameters are still valid and sensible in terms of magnitude.

I guess in the case where I'd like to sample e.g. a quantized learning rate with `tune.qloguniform(1e-4, 1e-1, 5e-4)` and potentially pass this to different search algorithms, I wouldn't want to change this configuration specifically to `tune.loguniform(1e-4, 1e-1)` for Optuna.